specs/core/1_new_shards.md

```diff
@@ -0,0 +1,546 @@
+    - [Helpers](#helpers)
+        - [`get_online_validators`](#get_online_validators)
+        - [`pack_compact_validator`](#pack_compact_validator)
+        - [`committee_to_compact_committee`](#committee_to_compact_committee)
+        - [`get_light_client_committee`](#get_light_client_committee)
+        - [`get_indexed_attestation`](#get_indexed_attestation)
+        - [`update_gasprice`](#update_gasprice)
+        - [`is_valid_indexed_attestation`](#is_valid_indexed_attestation)
+        - [`get_attestation_shard`](#get_attestation_shard)
+    - [Beacon Chain Changes](#beacon-chain-changes)
+        - [New beacon state fields](#new-beacon-state-fields)
+        - [New beacon block data fields](#new-beacon-block-data-fields)
+        - [Attestation processing](#attestation-processing)
+            - [`validate_attestation`](#validate_attestation)
+            - [`apply_shard_transition`](#apply_shard_transition)
+            - [`process_attestations`](#process_attestations)
+        - [Misc block post-processing](#misc-block-post-processing)
+        - [Light client processing](#light-client-processing)
+        - [Epoch transition](#epoch-transition)
+    - [Fraud proofs](#fraud-proofs)
+    - [Shard state transition function](#shard-state-transition-function)
+    - [Honest committee member behavior](#honest-committee-member-behavior)
+
+<!-- /TOC -->
+
+## Introduction
+
+This document describes the shard transition function (data layer only) and the shard fork choice rule as part of Phase 1 of Ethereum 2.0.
+
+## Configuration
+
+### Misc
+
+| Name | Value | Unit | Duration |
+| - | - | - | - | 
+| `MAX_SHARDS` | `2**10` (= 1024) |
+| `ACTIVE_SHARDS` | `2**6` (= 64) |
+| `ONLINE_PERIOD` | `2**3` (= 8) | epochs | ~51 min |
+| `LIGHT_CLIENT_COMMITTEE_SIZE` | `2**7` (= 128) |
+| `LIGHT_CLIENT_COMMITTEE_PERIOD` | `2**8` (= 256) | epochs | ~27 hours |
+| `SHARD_BLOCK_CHUNK_SIZE` | `2**18` (= 262,144) | |
+| `MAX_SHARD_BLOCK_CHUNKS` | `2**2` (= 4) | |
+| `BLOCK_SIZE_TARGET` | `3 * 2**16` (= 196,608) | |
+| `SHARD_BLOCK_OFFSETS` | `[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233]` | |
+| `MAX_SHARD_BLOCKS_PER_ATTESTATION` | `len(SHARD_BLOCK_OFFSETS)` | |
+| `EMPTY_CHUNK_ROOT` | `hash_tree_root(BytesN[SHARD_BLOCK_CHUNK_SIZE]())` | |
+| `MAX_GASPRICE` | `2**14` (= 16,384) | Gwei | |
+| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `2**3` (= 8) | |
+| `DOMAIN_SHARD_LIGHT_CLIENT` | `192` | |
+| `DOMAIN_SHARD_PROPOSAL` | `193` | |
+
+## Containers
+
+### `ShardBlockWrapper`
+
+```python
+class ShardBlockWrapper(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body: BytesN[SHARD_BLOCK_CHUNK_SIZE]
+    signature: BLSSignature
+```
+
+### `ShardSignedHeader`
+
+```python
+class ShardSignedHeader(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body_root: Hash
+```
+
+### `ShardState`
+
+```python
+class ShardState(Container):
+    slot: Slot
+    gasprice: Gwei
```

Comment:
Need to set initial `gasprice` value.

Reply:
I think no need. It's zero by default, but then after one slot `get_updated_gasprice` pushes it up to `MIN_GASPRICE`.